fix(actions): send error HTTP header from action forward hook

hypeJunction opened the pull request:
ajax_forward_hook() now sends error HTTP header, if forward reason is an
HTTP error code. This makes sure that response object returned by non-action
URLs is not output as raw string to the user (e.g. when requested by lightbox)

Fixes #9027

mrclay commented on engine/classes/Elgg/ActionsService.php:
```diff
@@ -322,7 +322,24 @@ public function ajaxForwardHook($hook, $reason, $return, $params) {
+			);
+
+			if (isset($httpCodes[$reason])) {
+				header("HTTP/1.1 $reason {$httpCodes[$reason]}", true);
```
But we'd still send the response?

mrclay replied:
I guess it doesn't matter either way.

hypeJunction replied:
I don't think it matters. I suppose sending it would allow someone to parse JSON from response body, if they must.

mrclay commented:
"an error" instead of "and error". This seems safe to me as successful handlers shouldn't forward with errors as the reason.

mrclay commented:
> elgg.ajax should be able to interpret it as an error, and lightbox will fail silently.

Did you actually test this?

hypeJunction commented:
`elgg.ajax` displays a generic AJAX error message.

mrclay commented:
LGTM, @juho-jaakkola what you think?

hypeJunction commented:
If there is another LGTM I will resubmit without whitespace noise, and http codes should be a constant in one of the servicea., or better yet a method that calls a hook.

juho-jaakkola commented:
LGTM

mrclay commented:
Just remove the whitespace noise for 1.12. We don't really want refactoring there.

hypeJunction committed:
fix(actions): send error HTTP header from action forward hook

ajax_forward_hook() now sends error HTTP header, if forward reason is an
HTTP error code. This makes sure that response object returned by non-action
URLs is not output as raw string to the user (e.g. when requested by lightbox)

Fixes #9027

hypeJunction commented:
@mrclay Exactly what I was thinking. Done.
